spec/datadog/core/remote/component_spec.rb

```diff
@@ -0,0 +1,212 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'datadog/core/remote/component'
+
+RSpec.describe Datadog::Core::Remote::Component do
+  let(:settings) { Datadog::Core::Configuration::Settings.new }
+  let(:agent_settings) { Datadog::Core::Configuration::AgentSettingsResolver.call(settings, logger: nil) }
+
+  describe '#initialize' do
+    subject(:component) { described_class.new(settings, agent_settings) }
+
+    let(:transport_v7) { double }
+    let(:client) { double }
+    let(:worker) { component.instance_eval { @worker } }
```

Comment:
Minor: You may want to consider making the `Worker` a dependency that can be injected. This way it would make the class more testable (because you can inject a custom implementation, instead of having to "grab" the internal state of the class to get the one that one created), as well as more flexible -- in the future we could reuse the class with a different worker with zero changes!